File: src/main/scala/part2/TileButton.scala
package part2

import java.awt.Color
import java.awt.event.{MouseAdapter, MouseEvent}
import javax.swing.{BorderFactory, ImageIcon, JButton}

case class TileButton(tile: Tile) extends JButton(new ImageIcon(tile.getImage)) {
  addMouseListener(new MouseAdapter {
    override def mouseClicked(e: MouseEvent): Unit =
      setBorder(BorderFactory.createLineBorder(Color.red, 2))
  })
}

File: src/main/java/part2/rmi/TileButton.java
package part2.rmi;

import javax.swing.*;
import java.awt.*;
import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;

public class TileButton extends JButton{

	private boolean isSelected = false;

	public TileButton(final Tile tile) {
		super(new ImageIcon(tile.getImage()));
		
		addMouseListener(new MouseAdapter() {            
            @Override
            public void mouseClicked(MouseEvent e) {
            	isSelected = !isSelected;
            	if (isSelected)
            		setBorder(BorderFactory.createLineBorder(Color.red));
        		else
					setBorder(BorderFactory.createLineBorder(Color.gray));

            }
        });
	}
}
